todo_service/repo/todo.go

```diff
@@ -102,6 +103,37 @@ func (tr *TodoRepo) GetForUser(id int64, userID int64) (models.Todo, error) {
 	}
 }
 
+func (tr *TodoRepo) GetByDate(dueDate string) ([]int64, error) {
+	var userIDCollection []int64
+	rows, err := tr.db.Query(`select DISTINCT l.user_id from todos t 
+	INNER JOIN list l ON l.id=t.list_id 
+	WHERE date_trunc('day',t.due_date) = $1`, dueDate)
+
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var userID int64
+
+		err := rows.Scan(&userID)
+		if err != nil {
+			return nil, err
+		}
+		userIDCollection = append(userIDCollection, userID)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("database iteration error: %v", err)
+	}
+
+	if len(userIDCollection) == 0 {
+		return nil, fmt.Errorf("no users found with task due %s", dueDate)
+	}
```

Comment:
If not such collection is found then API will fail?
Ideally it can return just an empty list